File: ios/VietMapNavigationModule.m
// RCTCalendarModule.m
#import "VietMapNavigationModule-Header.h"
#import <React/RCTLog.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(VietMapNavigationModule, NSObject)
RCT_EXTERN_METHOD(buildRoute:(NSArray *)location profile:(NSString *)profile)
RCT_EXTERN_METHOD(startNavigation)
RCT_EXTERN_METHOD(finishNavigation)
RCT_EXTERN_METHOD(overView)
RCT_EXTERN_METHOD(cancelNavigation)
RCT_EXTERN_METHOD(clearRoute)
RCT_EXTERN_METHOD(recenter)
RCT_EXTERN_METHOD(startSpeedAlert)
RCT_EXTERN_METHOD(stopSpeedAlert)
RCT_EXTERN_METHOD(isSpeedAlertActive:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(configureAlertAPI:(NSString *)apiKey apiID:(NSString *)apiID)
RCT_EXTERN_METHOD(configVehicleSpeedAlert:(NSString *)vehicleId vehicleType:(NSInteger)vehicleType seats:(NSInteger)seats weight:(double)weight)

@end

File: ios/VietMapNavigationModule.swift
import Foundation
@objc(VietMapNavigationModule)
class VietMapNavigationModule: NSObject {
    @objc
    func buildRoute(_ location: NSArray, profile: NSString) {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.buildRoute(location: location, profile: profile)
        }
    }
    
    @objc
    func startNavigation() {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.startNavigation()
        }
    }
    
    @objc
    func finishNavigation() {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.finishNavigation()
        }
    }
    
    @objc
    func overView() {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.overView()
        }
    }
    
    @objc
    func cancelNavigation() {
        // print("cancelNavigation hereeeeeee")
    }
    
    @objc
    func clearRoute() {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.clearRoute()
        }
    }
    
    @objc
    func recenter() {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.recenter()
        }
    }
    
    @objc
    func startSpeedAlert() {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.startSpeedAlert()
        }
    }
    
    @objc
    func stopSpeedAlert() {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.stopSpeedAlert()
        }
    }
    
    @objc
    func isSpeedAlertActive(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            let isActive = map.isSpeedAlertActive()
            resolve(isActive)
        }
    }
    
    @objc
    func configureAlertAPI(_ apiKey: String, apiID: String) {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.apiKeyAlert = apiKey
            map.apiIDAlert = apiID
        }
    }
    
    @objc
    func configVehicleSpeedAlert(_ vehicleId: String, vehicleType: Int, seats: Int, weight: Double) {
        DispatchQueue.main.async {
            let map = VietMapNavigationView.shared
            map.configVehicleSpeedAlert(vehicleId: vehicleId, vehicleType: vehicleType, seats: seats, weight: weight)
        }
    }
}
